Initial overhaul of NuCypher Official Docs (#2870)
nucypher/nucypher: docs/source/pre_application/node_requirements.rst

=================

Minimum System Requirements
---------------------------

* Debian/Ubuntu (Recommended)
* 20GB storage
* 4GB RAM
* x86 architecture
* Static IP address
* Exposed TCP port 9151

Nodes can be run on cloud infrastructure - for example, a
`Digital Ocean 4GB Basic Droplet <https://www.digitalocean.com/pricing/>`_
satisfies the requirements listed above.

The above requirements only apply if you intend to run a node yourself.
Alternatively, a staking provider can run a node on your behalf.

PRE Node Configuration Requirements
-----------------------------------

Ethereum Node Provider URI
++++++++++++++++++++++++++

The PRE node will need to execute an initial transaction to confirm its
operation on startup, and therefore requires access to an
`ethereum node <https://web3py.readthedocs.io/en/stable/node.html>`_
(either a local or remote ethereum node) to broadcast the transaction. Remote
ethereum providers include Infura, Alchemy etc. and an HTTPS URI will need to
be configured (``https://<URI>``), whereas a local full node would entail running
`geth <https://geth.ethereum.org/>`_ locally and configured using the
IPC URI (``ipc://<PATH TO IPC FILE>``).

> This value will be provided via the ``--provider`` CLI parameter.

Reviewer: ```suggestion
This value is given via the ``--eth-provider`` CLI parameter.
```

Author: ✔️
